dev/dialects/db2/README.md

```diff
@@ -0,0 +1,84 @@
+# DB2 README
+
+There are many versions of DB2 products, which have different features and different documentation. For instance, in this version `JSON_ARRAYAGG` and `ARRAY_AGG` functions are not available, only `LISTAGG` is.  To confirm this query can be run:
+
+   ```bash
+   db2 "select distinct routinename from syscat.routines where routinename like 'JSON%' or routinename like '%ARRAY%' or routinename like '%AGG%' order by 1"
+   ```
+
+Types of DB2 Platforms
```

Comment:
Is the DB2 dialect supporting all 4?

Reply:
No idea. I was surprised there was a community edition.  Per something I read somewhere in Sequelize comments or maybe Contributing, it seems like its leaning on the community for supporting some of the features, for platforms that may not be broadly available to the public.

I would think the community edition is the lowest common denominator and that supporting it would support the other, but that's a guess.  If you want the others, you have to pay.